platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ZipUtils.java

```diff
@@ -0,0 +1,284 @@
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.apache.commons.io.FileUtils.writeStringToFile;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZipUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ZipUtils.class);
+
+  private static final String IMPORT_KEY_PADDING_ERROR =
+      "Import error: invalid padding used for import key {}.";
+
+  private static final String IMPORT_KEY_ALGORITHM_ERROR =
+      "Import error: invalid algorithm used for import key {}.";
+
+  private static final String EXPORT_KEY_PADDING_ERROR =
+      "Export error: invalid padding used for export key";
+
+  private static final String EXPORT_KEY_ALGORITHM_ERROR =
+      "Export error: invalid algorithm used for export key";
+
+  @VisibleForTesting static final String KEY_ALGORITHM = "AES";
+
+  @VisibleForTesting static final String CIPHER_ALGORITHM = "AES/ECB/PKCS5Padding";
+
+  @VisibleForTesting static final String DECRYPTION_KEY_FILE_NAME = "export.key";
+
+  @VisibleForTesting static final String CHECKSUM_FILE_EXTENSION = ".sha256";
+
+  @VisibleForTesting static final String CHECKSUM_DIGEST_ALGORITHM = "SHA-256";
+
+  /**
+   * Streams the contents of a {@link ZipInputStream} as a {@link Map} where the content of each
+   * {@link ZipEntry} is stored inside a key corresponding to each {@link ZipEntry}'s name
+   *
+   * @param in the {@link ZipInputStream} to retrieve the entries from
+   * @return the stream of entries
+   */
+  public static Map<ZipEntry, byte[]> entriesFrom(ZipInputStream in) {
+    ZipEntry entry;
+    Map<ZipEntry, byte[]> entries = new HashMap<>();
+    try {
+      while ((entry = in.getNextEntry()) != null) {
+        if (!entry.isDirectory()) {
+          byte[] content = IOUtils.toByteArray(in);
```

Comment:
❓ Do we have a ballpark figure on how large the memory footprint of this map will be during its lifetime?

Reply:
No bigger than it was when using the ZipFile. But as big as the number of files we export. The biggest zip file I saw so far was about 200KB